src/components/react-json-schemaForm/TableField/TableField.js

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import Thead from './Thead';
+import Tbody from './Tbody';
+
+import './styles.scss';
+
+const TableField = props => {
+  const {
+    formData,
+    onChange,
+    schema: { title },
+  } = props;
+
+  const [data, setData] = useState(formData);
+
+  useEffect(() => {
+    onChange(data);
+  }, [data, onChange]);
+
+  const handleChange = useCallback((row, key) => value => {
+    const nextData = [...data];
+    nextData[row][key] = value;
+    onChange(nextData);
+  }, [data, onChange]);
+
+  return (
+    <fieldset className="field field-array field-array-of-object">
+      <legend>{title}</legend>
+      <div className="details__table tableField">
+        <table>
+          <Thead {...props} />
+          <Tbody
+            {...props}
+            formData={data}
+            onChange={handleChange}
+            onUpdatingRow={setData}
+          />
+        </table>
+      </div>
+    </fieldset>
+  );
+};
```

Comment:
The reason why *original* `props` are used for `<Thead />` but  *altered* props in `<Tbody />` is not clearly understandable. I don't say this has to be changed or is wrong. But it adds some unecessary complexity.

I would maybe have done a first wrapping (dump) stateless component with `<fieldset />`, `title`, `<table />` and `<Thead {...props} />`, then, another component holding state and `handleChange` method, and called it in place of `<Tbody />`.


```jsx
const CustomTbody = ({ formData, onChange, ...props }) => {
  const [data, setData] = useState(formData);

  useEffect(() => {
    onChange(data);
  }, [data, onChange]);

  const handleChange = useCallback((row, key) => value => {
    const nextData = [...data];
    nextData[row][key] = value;
    onChange(nextData);
  }, [data, onChange]);

  return (
    <Tbody
      formData={data}
      onChange={handleChange}
      onUpdatingRow={setData}
      {...props}
    />
  );
};


const TableField = props => (
  <fieldset className="field field-array field-array-of-object">
    <legend>{title}</legend>
    <div className="details__table tableField">
      <table>
        <Thead {...props} />
        <CustomTbody {...props} />
      </table>
    </div>
  </fieldset>
);
```

Reply:
Right, 
Here, there was some logic which shouldn't be.
 i just moved the `useEffect` and the `handlechange` directly in the `Tbody` component